ravenframework/Models/PostProcessors/SparseSensing.py

```diff
@@ -0,0 +1,190 @@
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, optional, dictionary with initialization options
+      @ Out, None
+    """
+    super().initialize(runInfo, inputs, initDict)
+    if len(inputs)>1:
+      self.raiseAnError(IOError, 'Post-Processor', self.name, 'accepts only one dataObject')
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    self.name = paramInput.parameterValues['name']
+    for child in paramInput.subparts:
+      self.sparseSensingGoal = child.parameterValues['subType']
+      self.nSensors = child.findFirst('nSensors').value
+      self.nModes = child.findFirst('nModes').value
+      self.basis = child.findFirst('basis').value
+      self.sensingFeatures = child.findFirst('features').value
+      self.sensingTarget = child.findFirst('target').value
+      self.optimizer = child.findFirst('optimizer').value
+      if child.findFirst('seed') is not None:
+        self.seed = child.findFirst('seed').value
+      else:
+        self.seed = None
+      if child.parameterValues['subType'] not in self.goalsDict.keys():
+        self.raiseAnError(IOError, '{} is not a recognized option, allowed options are {}'.format(child.getName(),self.goalsDict.keys()))
+    _, notFound = paramInput.subparts[0].findNodesAndExtractValues(['nModes','nSensors','features','target'])
+    # notFound must be empty
+    assert(not notFound)
+
+  def run(self,inputIn):
+    """
+      @ In, inputIn, dict, dictionaries which contains the data inside the input DataObjects
+      @ Out, outputDic, dict, dictionary which contains the data to be collected by output DataObject
+    """
+    _, _, inputDS = inputIn['Data'][0]
+
+    # #identify features
+    self.features = self.sensingFeatures
+    #don't keep the pivot parameter in the feature space
+    if self.pivotParameter in self.features:
+      self.features.remove(self.pivotParameter)
+    if self.basis.lower() == 'svd':
+      basis=ps.basis.SVD(n_basis_modes=self.nModes)
+    elif self.basis.lower() == 'identity':
+      basis=ps.basis.Identity(n_basis_modes=self.nModes)
+    elif self.basis.lower() == 'randomprojection':
+      basis=ps.basis.RandomProjection(n_basis_modes=self.nModes)
+    else:
+      self.raiseAnError(IOError, 'basis are not recognized')
+
+    if self.optimizer.lower() == 'qr':
+      optimizer = ps.optimizers.QR()
+    else:
+      self.raiseAnError(IOError, 'optimizer {} not implemented!!!'.format(self.optimizer))
+
+    model = ps.SSPOR(basis=basis,n_sensors = self.nSensors,optimizer = optimizer)
+
+    features = {}
+    for var in self.sensingFeatures:
+      features[var] = np.atleast_1d(inputDS[var].data)
+
+    data = inputDS[self.sensingTarget].data
+    ## TODO: add some assertions to check the shape of the data matrix in case of steady state and time-dependent data
```

Comment:
Can you check the checks in the "TODO"?

Reply:
@wangcj05, this is for the case of time-dependent data, which is not done yet. I left this as a reminder for Josh when he implements it.

I have added another assert here for the steady state case, yet I don't think it is possible to make it false.